### PR TITLE
[Rule Tuning] Review and Tune `Potential Malicious File Downloaded from Google Drive`

### PR DESCRIPTION
**Pull request by terrancedejesus**
## Summary
This PR removes benign programming executables as well as signed and trusted binaries from Windows, macOS and Linux. Tuning reduces global alert metrics from ~175,225 to ~900 in a 90 day period.

### CODE REVIEW

File: rules/cross-platform/command_and_control_google_drive_malicious_file_download.toml [hunk 1]
```diff
@@ -38,27 +38,49 @@ sequence by host.id, process.entity_id with maxspan=30s
...
+    and not process.executable:
+        ("/bin/terraform",
+        "*/bin/dockerd",
+        "/usr/local/bin/docker-init",
+        "*/bin/go",
+        "?:\\Program Files*\\Mozilla Firefox\firefox.exe",
+        "?:\\Program Files*\\Google\\Chrome\\Application\\chrome.exe")
+
+    /* ignore shellscripts + go install from legitimate repository*/
+    and not ( process.executable == "/bin/sh" and process.args : "go install google.golang.org*")]
```
Comment (Aegrah) [inline rules/cross-platform/command_and_control_google_drive_malicious_file_download.toml:70]:
```suggestion
    and not (process.executable == "/bin/sh" and process.args : "go install google.golang.org*")]
```

**Review by Aegrah — APPROVED** [review]
Very nice! Glad to see this rule receiving some great tuning

File: rules/cross-platform/command_and_control_google_drive_malicious_file_download.toml [hunk 2]
```diff
@@ -38,27 +38,49 @@ sequence by host.id, process.entity_id with maxspan=30s
 
     /* Look for Google Drive download URL with AV flag skipping */
     (process.args : "*drive.google.com*" and process.args : "*export=download*" and process.args : "*confirm=no_antivirus*")
-]
+
+    /* ignore trusted processes */
+    and not (
+        process.code_signature.trusted == true or
+        process.code_signature.subject_name:
```
[the remark this answers is not in the document]
Comment (Samirbous) [reply rules/cross-platform/command_and_control_google_drive_malicious_file_download.toml:45]: ++

File: rules/cross-platform/command_and_control_google_drive_malicious_file_download.toml [hunk 3]
```diff
@@ -38,27 +38,49 @@ sequence by host.id, process.entity_id with maxspan=30s
...
-        "curl", "wget", "scp", "ftp", "python", "perl", "ruby"))] */
-]
+    (destination.as.organization.name : "GOOGLE" and event.action == "connection_attempted")]
 
 /* Identify the creation of files following Google Drive connection with extensions commonly used for executables or libraries */
-[file where event.action == "creation" and file.extension : (
-    "exe", "dll", "scr", "jar", "pif", "app", "dmg", "pkg", "elf", "so", "bin", "deb", "rpm","sh","hta","lnk"
-    )
-]
+[file where event.action == "creation" and not process.code_signature.trusted == true and
```
Comment (w0rk3r) [inline rules/cross-platform/command_and_control_google_drive_malicious_file_download.toml:80]:
```suggestion
[file where event.action == "creation" and
```
I don't think we need this one, as they are already excluded from the first sequence event, and would exclude any signed process

**Review by w0rk3r — APPROVED** [review]
🚀 🧙🏼

**Pull request merged**